src/shogun/kernel/SpectralMixtureKernel.cpp

```diff
@@ -0,0 +1,221 @@
+/*
+ * This software is distributed under BSD Clause 3 license (see LICENSE file).
+ *
+ * Written (W) 2015 Esben Soerig
+ */
+
+#include <shogun/kernel/SpectralMixtureKernel.h>
+
+using namespace shogun;
+
+CSpectralMixtureKernel::CSpectralMixtureKernel() : CDotKernel()
+{
+	/* We can't really say anything about number of components at this point
+	   We have to leave the object in an unusable state until a set of feature
+	   has been received
+	   */
+	init();
+}
+
+CSpectralMixtureKernel::CSpectralMixtureKernel(CDotFeatures* l, CDotFeatures* r,
+	int32_t num_components, int32_t size) : CDotKernel(size)
+{
+	init();
+	CDotKernel::init(l, r);
+	init_components(num_components);
+}
+
+CSpectralMixtureKernel::CSpectralMixtureKernel(const SGVector<float64_t>& weights,
+	const SGVector<float64_t>& means, const SGVector<float64_t>& stddeviations,
+	int32_t size) : CDotKernel(size)
+{
+	init();
+	set_component_parameters(weights, means, stddeviations);
+}
+
+void CSpectralMixtureKernel::init()
+{
+	SG_ADD(&m_weights, "weights",
+		"Vector of weights for each mixture component", MS_AVAILABLE, GRADIENT_AVAILABLE);
+	SG_ADD(&m_means, "means",
+		"Vector of means for each mixture component", MS_AVAILABLE, GRADIENT_AVAILABLE);
+	SG_ADD(&m_stddeviations, "stddeviations",
+		"Vector of standard deviations for each mixture component", MS_AVAILABLE, GRADIENT_AVAILABLE);
+}
+
+bool CSpectralMixtureKernel::init_components(int32_t num_components)
+{
+	REQUIRE(has_features(), "Features not set - cannot initialize components.");
+
+	m_weights=SGVector<float64_t>(num_components);
+	m_weights.set_const(1.0/num_components);
+
+	m_means=SGVector<float64_t>(num_components);
+	m_means.set_const(0.0);
+
+	m_stddeviations=SGVector<float64_t>(num_components);
+	m_stddeviations.set_const(1.0);
+
+	return true;
+}
+
+int32_t CSpectralMixtureKernel::get_num_components() const
+{
+	return m_weights.size();
+}
+
+void CSpectralMixtureKernel::set_component_parameters(const SGVector<float64_t>& weights,
+	const SGVector<float64_t>& means, const SGVector<float64_t>& stddeviations)
+{
+	bool same_size= (weights.size()==means.size()) && (weights.size()==stddeviations.size());
+	REQUIRE(same_size, "Weights, means, and stddeviations vectors must be of equal size.");
```

Comment:
Could you be more explicit here?
I.e. state the numbers which dont match, this can be done via multipe REQUIRE statements